### PR TITLE
Handle TypedParamStrings

### PR DESCRIPTION
**Pull request by trapgate**
Add handling for TypedParamStrings, which are different from the other
TypedParam... types in that the decoded value is variable-sized.

### CODE REVIEW

File: libvirt.go [hunk 1]
```diff
@@ -1347,14 +1411,15 @@ func (l *Libvirt) SetBlockIOTune(dom string, disk string, limits ...BlockLimit)
...
+		payload.Params = append(payload.Params, tp)
 	}
 
+	fmt.Println(payload)
 	buf, err := encode(&payload)
 	if err != nil {
 		return err
 	}
-
+	fmt.Println(buf)
```
Comment (benlemasurier) [inline libvirt.go]: oops?

File: libvirt.go [hunk 2]
```diff
@@ -1403,20 +1468,50 @@ func (l *Libvirt) GetBlockIOTune(dom string, disk string) ([]BlockLimit, error)
 		return nil, decodeError(r.Payload)
 	}
 
-	dec := xdr.NewDecoder(bytes.NewReader(r.Payload))
-	result := struct {
-		Limits     []TypedParam
-		ParamCount uint32
-	}{}
-	_, err = dec.Decode(&result)
+	fmt.Println(r.Payload)
```
Comment (benlemasurier) [inline libvirt.go]: oops?

Comment (trapgate) [reply libvirt.go]: Yep, fixed. Also added a comment to the bytes that add a test for strings.

File: libvirt.go [hunk 3]
```diff
@@ -1347,14 +1411,15 @@ func (l *Libvirt) SetBlockIOTune(dom string, disk string, limits ...BlockLimit)
 
 	for _, limit := range limits {
 		tp := NewTypedParamULongLong(limit.Name, limit.Value)
-		payload.Params = append(payload.Params, *tp)
+		payload.Params = append(payload.Params, tp)
 	}
 
+	fmt.Println(payload)
```
Comment (benlemasurier) [inline libvirt.go]: this one too?

Comment (trapgate) [reply libvirt.go]: aye

File: libvirt.go (hunk 2 above)
Comment (benlemasurier) [reply libvirt.go]: I think you still missed this one

**Comment by benlemasurier** [comment]
:shipit:

**Pull request merged**